src/components/global/PlaygroundTabs/index.tsx

```diff
@@ -0,0 +1,231 @@
+
+// A very rough duck type, but good enough to guard against mistakes while
+// allowing customization
+function isTabItem(comp: ReactElement): comp is ReactElement<TabItemProps & { icon?: JSX.Element }> {
+  return typeof comp.props.value !== 'undefined';
+}
+
+function TabsComponent(props: Props): JSX.Element {
+  const { lazy, block, defaultValue: defaultValueProp, values: valuesProp, groupId, className } = props;
+  const [leftNavVisible, setLeftNavVisible] = useState(false);
+  const [rightNavVisible, setRightNavVisible] = useState(false);
+  const tabsNavEl = createRef<HTMLUListElement>();
+  const children = React.Children.map(props.children, (child) => {
+    if (isValidElement(child) && isTabItem(child)) {
+      return child;
+    }
+    // child.type.name will give non-sensical values in prod because of
+    // minification, but we assume it won't throw in prod.
+    throw new Error(
+      `Docusaurus error: Bad <Tabs> child <${
+        // @ts-expect-error: guarding against unexpected cases
+        typeof child.type === 'string' ? child.type : child.type.name
+      }>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`
+    );
+  });
+  const values =
+    valuesProp ??
+    // Only pick keys that we recognize. MDX would inject some keys by default
+    children.map(({ props: { value, label, icon, attributes } }) => ({
+      value,
+      label,
+      attributes,
+      icon,
+    }));
+  const dup = duplicates(values, (a, b) => a.value === b.value);
+  if (dup.length > 0) {
+    throw new Error(
+      `Docusaurus error: Duplicate values "${dup
+        .map((a) => a.value)
+        .join(', ')}" found in <Tabs>. Every value needs to be unique.`
+    );
+  }
+  // When defaultValueProp is null, don't show a default tab
+  const defaultValue =
+    defaultValueProp === null
+      ? defaultValueProp
+      : defaultValueProp ?? children.find((child) => child.props.default)?.props.value ?? children[0]?.props.value;
+  if (defaultValue !== null && !values.some((a) => a.value === defaultValue)) {
+    throw new Error(
+      `Docusaurus error: The <Tabs> has a defaultValue "${defaultValue}" but none of its children has the corresponding value. Available values are: ${values
+        .map((a) => a.value)
+        .join(', ')}. If you intend to show no default tab, use defaultValue={null} instead.`
+    );
+  }
+
+  const [selectedValue, setSelectedValue] = useState(defaultValue);
+  const tabRefs: (HTMLLIElement | null)[] = [];
+  const { blockElementScrollPositionUntilNextRender } = useScrollPositionBlocker();
+
+  const handleTabChange = (event: React.FocusEvent<HTMLLIElement> | React.MouseEvent<HTMLLIElement>) => {
+    const newTab = event.currentTarget;
+    const newTabIndex = tabRefs.indexOf(newTab);
+    const newTabValue = values[newTabIndex]!.value;
+
+    if (newTabValue !== selectedValue) {
+      blockElementScrollPositionUntilNextRender(newTab);
```

Comment:
Why is this needed?

Reply:
This is taken from the docusaurus original source: https://github.com/facebook/docusaurus/blob/4e45e14fdd760a7eff13f842c81654eba5e872a3/packages/docusaurus-theme-common/src/utils/scrollUtils.tsx#L178-L188

In theory, since we aren't using `groupId` in this feature set, it could be removed, but I sided on not removing any unnecessary code from the original implementation in the event we wanted to pull new changes from their code.